### PR TITLE
chore: upgrade controller-runtime

### PR DESCRIPTION
**Pull request by bernot-dev**
Upgrade controller-runtime to v0.15.x

This will unblock #663, as previous versions of controller-runtime do not correctly support providing your own `GetCertificate` method to provide certs.

### CODE REVIEW

**Comment by bernot-dev** [comment]
@TheSpiritXIII #669 may resolve the `validate` check failure. Unit test failures still need to be debugged.

**Review by TheSpiritXIII — CHANGES_REQUESTED** [review]
Thanks for the good work!

File: e2e/authorization_test.go [hunk 1]
```diff
@@ -33,7 +33,6 @@ import (
 )
 
 func TestTLS(t *testing.T) {
-	t.Parallel()
```
[the remark this answers is not in the document]
Comment (bernot-dev) [reply e2e/authorization_test.go:36]: This test was passing for me locally when the Parallel was removed. But apparently it doesn't work in CI.

**Review by TheSpiritXIII — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**